File: InPlan.Web/Modules/Parametros/Tecnologias/TecnologiasForm.cs
using Serenity.ComponentModel;

namespace InPlan.Parametros.Forms;

[FormScript("Parametros.Tecnologias")]
[BasedOnRow(typeof(TecnologiasRow), CheckNames = true)]
public class TecnologiasForm
{
    [HalfWidth]
    public int IdEmpresa { get; set; }
    [HalfWidth]
    public int IdArea { get; set; }
    [HalfWidth]
    public string Descripcion { get; set; }
    [TextAreaEditor(Rows = 5)]
    public string Observaciones { get; set; }
    [HideOnInsert]
    public bool Baja { get; set; }
}

File: InPlan.Web/Modules/ServerTypes/Parametros/TecnologiasForm.ts
import { LookupEditor, StringEditor, TextAreaEditor, BooleanEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface TecnologiasForm {
    IdEmpresa: LookupEditor;
    IdArea: LookupEditor;
    Descripcion: StringEditor;
    Observaciones: TextAreaEditor;
    Baja: BooleanEditor;
}

export class TecnologiasForm extends PrefixedContext {
    static formKey = 'Parametros.Tecnologias';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!TecnologiasForm.init)  {
            TecnologiasForm.init = true;

            var w0 = LookupEditor;
            var w1 = StringEditor;
            var w2 = TextAreaEditor;
            var w3 = BooleanEditor;

            initFormType(TecnologiasForm, [
                'IdEmpresa', w0,
                'IdArea', w0,
                'Descripcion', w1,
                'Observaciones', w2,
                'Baja', w3
            ]);
        }
    }
}